pandas/core/generic.py

```diff
@@ -3971,6 +3971,9 @@ def asof(self, where, subset=None):
         if not isinstance(where, Index):
             where = Index(where) if is_list else Index([where])
 
+        if self.isnull().values.all():
+            return pd.Series([np.nan])
```

Comment:
This will not work for a DataFrame (I mean: it will not be the correct return value, see the docstring)

Further, you can put this after the next line where `nulls` is defined and reuse that